apps/src/code-studio/pd/workshop_dashboard/reports/foorm/section_results.jsx

```diff
@@ -0,0 +1,106 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ChoiceResponses from '../../components/survey_results/choice_responses';
+import TextResponses from '../../components/survey_results/text_responses';
+import _ from 'lodash';
+
+export default class SectionResults extends React.Component {
+  static propTypes = {
+    section: PropTypes.string.isRequired,
+    answers: PropTypes.object.isRequired,
+    questions: PropTypes.object.isRequired
+  };
+
+  renderQuestionResults(surveyQuestions, questionId, surveyId) {
+    const {answers, section} = this.props;
+    let question = surveyQuestions[questionId];
+    let answer = answers[surveyId][questionId];
+    if (!answer) {
+      return null;
+    }
+
+    if (['scale', 'singleSelect', 'multiSelect'].includes(question['type'])) {
+      // numRespondents will get either a value (for multiSelect) or undefined.
+      const numRespondents = answer.num_respondents;
+
+      // Make a copy of the answers without the num_respondents and other_answers fields.
+      const filteredAnswers = _.omit(answer, [
+        'num_respondents',
+        'other_answers'
+      ]);
+
+      let possibleAnswersMap = question['choices'];
+      return (
+        <ChoiceResponses
+          perFacilitator={section === 'facilitator'}
+          numRespondents={numRespondents}
+          question={question['title'] || questionId}
+          answers={filteredAnswers}
+          possibleAnswers={Object.keys(possibleAnswersMap)}
+          possibleAnswersMap={possibleAnswersMap}
+          answerType={question['type']}
+          otherText={question['other_text']}
+          otherAnswers={answer['other_answers']}
+        />
+      );
+    } else if (question['type'] === 'matrix') {
```

Comment:
What do you think of splitting these two renders into two separate React components?  That way, when viewing the React DOM, we'll be able to tell at a glance which path we went down...

Reply:
I think that could make sense for the single choice vs matrix choice, I will split those. But the third branch doesn't do any special logic and goes to a different component (text) so I don't think it makes much sense to wrap that